dynamo/preprocessing/Preprocessor.py

```diff
@@ -0,0 +1,518 @@
+        # TODO: if inplace in select_genes is True, the following subset is unnecessary.
+        adata._inplace_subset_var(adata.var["use_for_pca"])
+        self.sctransform(adata, **self.sctransform_kwargs)
+        self._pca(adata)
+
+        temp_logger.finish_progress(progress_name="preprocess by sctransform recipe")
+
+    def config_pearson_residuals_recipe(self, adata: AnnData):
+        self.filter_cells_by_outliers = None
+        self.filter_genes_by_outliers = None
+        self.normalize_by_cells = None
+        self.select_genes = select_genes_by_pearson_residuals
+        self.select_genes_kwargs = {"n_top_genes": 2000}
+        self.normalize_selected_genes = normalize_layers_pearson_residuals
+        # select layers in adata to be normalized
+        normalize_layers = DKM.get_raw_data_layers(adata)
+        self.normalize_selected_genes_kwargs = {"layers": normalize_layers, "copy": False}
+        self.pca_kwargs = {"pca_key": "X_pca", "n_pca_components": 50}
+        self.use_log1p = False
+
+    def preprocess_adata_pearson_residuals(
+        self, adata: AnnData, tkey: Optional[str] = None, experiment_type: Optional[str] = None
+    ):
+        """A pipeline proposed in Pearson residuals (Lause, Berens & Kobak, 2021).
+        Lause, J., Berens, P. & Kobak, D. Analytic Pearson residuals for normalization of single-cell RNA-seq UMI data. Genome Biol 22, 258 (2021). https://doi.org/10.1186/s13059-021-02451-7
+
+        Parameters
+        ----------
+        adata : AnnData
+        tkey : Optional[str], optional
+            time key, by default None
+        experiment_type : str, optional
+            experiment type of data, by default None
+        """
+        temp_logger = LoggerManager.gen_logger("preprocessor-sctransform")
+        temp_logger.log_time()
+        self.standardize_adata(adata, tkey, experiment_type)
+
+        self._select_genes(adata)
+        self._normalize_selected_genes(adata)
+        self._pca(adata)
+
+        temp_logger.finish_progress(progress_name="preprocess by pearson residual recipe")
+
+    def config_monocle_pearson_residuals_recipe(self, adata: AnnData):
+        self.config_monocle_recipe(adata)
+        # self.filter_cells_by_outliers = None
+        # self.filter_genes_by_outliers = None
+        self.normalize_by_cells = normalize_cell_expr_by_size_factors
+        self.select_genes = select_genes_by_pearson_residuals
+        self.select_genes_kwargs = {"n_top_genes": 2000}
+        self.normalize_selected_genes = normalize_layers_pearson_residuals
+
+        self.normalize_selected_genes_kwargs = {"layers": ["X"], "copy": False}
+
+        self.pca_kwargs = {"pca_key": "X_pca", "n_pca_components": 50}
+        self.use_log1p = False
+
+    def preprocess_adata_monocle_pearson_residuals(
+        self, adata: AnnData, tkey: Optional[str] = None, experiment_type: Optional[str] = None
+    ):
+        """
+        A combined pipeline of monocle and pearson_residuals.
```

Comment:
pearson residual is used for adata.X but monocle for spliced/unspliced layers, right?

Reply:
`preprocess_adata_monocle_pearson_residuals` does what you describe.  `pearson_residuals` recipe normalizes all raw data layers (new/total, spliced/unspliced, X). Users can specify what to normalize by speicfy `self.normalize_selected_genes_kwargs`

Reply:
ok, got it. this sounds good to me.